app/src/main/java/com/tpp/theperiodpurse/AppScreen.kt

```diff
@@ -1,47 +1,125 @@
+import com.google.android.gms.common.api.ApiException
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.auth.GoogleAuthProvider
+import com.tpp.theperiodpurse.ui.calendar.CalendarViewModel
 import com.tpp.theperiodpurse.ui.component.BottomNavigation
 import com.tpp.theperiodpurse.ui.component.FloatingActionButton
 import com.tpp.theperiodpurse.ui.onboarding.*
 import com.tpp.theperiodpurse.ui.symptomlog.LoggingOptionsPopup
-import com.tpp.theperiodpurse.ui.theme.ThePeriodPurseTheme
-import java.time.LocalDate
 import dagger.hilt.android.AndroidEntryPoint
+import java.time.LocalDate
 
 @AndroidEntryPoint
 class MainActivity : ComponentActivity() {
+
+    companion object {
+
+        const val RC_SIGN_IN = 100
+    }
+
+    private lateinit var mAuth: FirebaseAuth
+    private lateinit var googleSignInClient: GoogleSignInClient
+
     @RequiresApi(Build.VERSION_CODES.O)
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
+        mAuth = FirebaseAuth.getInstance()
+
+        val gso = GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+            .requestIdToken(getString(R.string.default_web_client_id))
+            .requestEmail()
+            .build()
+
+        googleSignInClient = GoogleSignIn.getClient(this, gso)
+
         setContent {
-            ThePeriodPurseTheme {
-                Application()
+            if (mAuth.currentUser == null) {
+                Application { signIn() }
+            } else {
+//                val user: FirebaseUser = mAuth.currentUser!!
+                Application(true) { signIn() }
+
+
+            }
+        }
+    }
+
+    private fun signIn() {
+        val signInIntent = googleSignInClient.signInIntent
+        startActivityForResult(signInIntent, RC_SIGN_IN)
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+
+        // result returned from launching the intent from GoogleSignInApi.getSignInIntent()
+        if (requestCode == RC_SIGN_IN) {
+            val task = GoogleSignIn.getSignedInAccountFromIntent(data)
+            val exception = task.exception
+            if (task.isSuccessful) {
+                try {
+                    // Google SignIn was successful, authenticate with Firebase
+                    val account = task.getResult(ApiException::class.java)!!
+                    firebaseAuthWithGoogle(account.idToken!!)
+                } catch (e: Exception) {
+                    // Google SignIn Failed
+                    Log.d("SignIn", "Google SignIn Failed")
+                }
+            } else {
+                Log.d("SignIn", exception.toString())
             }
         }
+
+    }
+
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun firebaseAuthWithGoogle(idToken: String) {
```

Comment:
We probably won't use firebase. Instead, we will use the Google Drive API. You can remove this for now and look into Google Drive API if you work on #80